sbb_newspapers_org_image/eynollah.py

```diff
@@ -1,306 +1,198 @@
-        return image_box, [box[1], box[1] + box[3], box[0], box[0] + box[2]]
-
-    def otsu_copy(self, img):
-        img_r = np.zeros(img.shape)
-        img1 = img[:, :, 0]
-        img2 = img[:, :, 1]
-        img3 = img[:, :, 2]
-        # print(img.min())
-        # print(img[:,:,0].min())
-        # blur = cv2.GaussianBlur(img,(5,5))
-        # ret3,th3 = cv2.threshold(blur,0,255,cv2.THRESH_BINARY+cv2.THRESH_OTSU)
-        retval1, threshold1 = cv2.threshold(img1, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)
-        retval2, threshold2 = cv2.threshold(img2, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)
-        retval3, threshold3 = cv2.threshold(img3, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)
-
-        img_r[:, :, 0] = threshold1
-        img_r[:, :, 1] = threshold1
-        img_r[:, :, 2] = threshold1
-        return img_r
-
-    def otsu_copy_binary(self,img):
-        img_r=np.zeros((img.shape[0],img.shape[1],3))
-        img1=img[:,:,0]
-
-        retval1, threshold1 = cv2.threshold(img1, 0, 255, cv2.THRESH_BINARY+cv2.THRESH_OTSU)
 
-        
+        self.model_dir_of_enhancemnet = dir_models + "/model_enhancement.h5"
+        self.model_dir_of_col_classifier = dir_models + "/model_scale_classifier.h5"
+        self.model_region_dir_p = dir_models + "/model_main_covid19_lr5-5_scale_1_1_great.h5"  # dir_models +'/model_main_covid_19_many_scalin_down_lr5-5_the_best.h5'#'/model_main_covid19_lr5-5_scale_1_1_great.h5'#'/model_main_scale_1_1und_1_2_corona_great.h5'
+        # self.model_region_dir_p_ens = dir_models +'/model_ensemble_s.h5'#'/model_main_covid19_lr5-5_scale_1_1_great.h5'#'/model_main_scale_1_1und_1_2_corona_great.h5'
+        self.model_region_dir_p2 = dir_models + "/model_main_home_corona3_rot.h5"
 
-        img_r[:,:,0]=threshold1
-        img_r[:,:,1]=threshold1
-        img_r[:,:,2]=threshold1
+        self.model_region_dir_fully_np = dir_models + "/model_no_patches_class0_30eopch.h5"
+        self.model_region_dir_fully = dir_models + "/model_3up_new_good_no_augmentation.h5"  # "model_3col_p_soft_10_less_aug_binarization_only.h5"
+
+        self.model_page_dir = dir_models + "/model_page_mixed_best.h5"
+        self.model_region_dir_p_ens = dir_models + "/model_ensemble_s.h5"  # dir_models +'/model_main_covid_19_many_scalin_down_lr5-5_the_best.h5' #dir_models +'/model_ensemble_s.h5'
+        ###self.model_region_dir_p = dir_models +'/model_layout_newspapers.h5'#'/model_ensemble_s.h5'#'/model_layout_newspapers.h5'#'/model_ensemble_s.h5'#'/model_main_home_5_soft_new.h5'#'/model_home_soft_5_all_data.h5' #'/model_main_office_long_soft.h5'#'/model_20_cat_main.h5'
+        self.model_textline_dir = dir_models + "/model_textline_newspapers.h5"  #'/model_hor_ver_home_trextline_very_good.h5'# '/model_hor_ver_1_great.h5'#'/model_curved_office_works_great.h5'
+
+    def predict_enhancement(self, img):
+        model_enhancement, session_enhancemnet = self.start_new_session_and_model(self.model_dir_of_enhancemnet)
```

Comment:
Maybe we can consider renaming some of the models or making the model names more consistent (good/best/very good/great) @vahidrezanezhad